src/app/util/ember-compatibility-functions.cpp

```diff
@@ -405,14 +422,30 @@ CHIP_ERROR stringTlvDataToAttributeBuffer(TLV::TLVReader & aReader, uint16_t & d
 
 CHIP_ERROR prepareWriteData(EmberAfAttributeType expectedType, TLV::TLVReader & aReader, uint16_t & dataLen)
 {
-    switch (BaseType(expectedType))
+    EmberAfAttributeType baseType;
+
+    baseType = BaseType(expectedType);
+    switch (baseType)
     {
     case ZCL_BOOLEAN_ATTRIBUTE_TYPE: // Boolean
         return numericTlvDataToAttributeBuffer<bool>(aReader, dataLen);
     case ZCL_INT8U_ATTRIBUTE_TYPE: // Unsigned 8-bit integer
         return numericTlvDataToAttributeBuffer<uint8_t>(aReader, dataLen);
     case ZCL_INT16U_ATTRIBUTE_TYPE: // Unsigned 16-bit integer
         return numericTlvDataToAttributeBuffer<uint16_t>(aReader, dataLen);
+    case ZCL_INT24U_ATTRIBUTE_TYPE: // Unsigned 24-bit integer
+    {
+        uint32_t value;
+        static_assert(3 <= sizeof(attributeData), "Value cannot fit into attribute data");
+        ReturnErrorOnFailure(aReader.Get(value));
+        if (value > static_cast<uint32_t>((1UL << (3 * 8)) - 1))
+        {
+            return CHIP_ERROR_INVALID_INTEGER_VALUE;
+        }
+        dataLen = emberAfGetDataSize(baseType);
+        emberAfCopyInt24u(attributeData, 0, value);
```

Comment:
OK, so this is in fact storing as little-endian, so I guess the question is just around default values.

Well, that and the oddity of storing these as little-endian while all other integers are stored native-endian...